### PR TITLE
[SW-850] Multisig: Add question-mark icon and info popup for the "x/y of z" consensus display

### PR DESCRIPTION
**Pull request by konradodo**

### CODE REVIEW

File: src/popup/locales/en.json [hunk 1]
```diff
@@ -163,6 +163,14 @@
...
       "syncing": "Syncing..."
     },
+    "consensus-info": {
+      "title": "Multisig consensus",
+      "subtitle": " X/Y of Z",
+      "first-point": "<strong>X =</strong> accounts that have already signed the transaction;",
+      "second-point": "<strong>Y =</strong> number of signatures required to approve the transaction;",
+      "third-point": "<strong>Z =</strong> number of authorized signers.",
+      "description": "If <strong>X=Y</strong> the multisig consensus has been reached. The transaction proposal has been approved and the transaction can be sent by any of the authorized signers."
+    },
```
Comment (peronczyk) [inline src/popup/locales/en.json:173]: Please use camelCase for the translation file keys.

File: src/popup/components/Modals/ConsensusInfo.vue [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        />
+      </div>
+    </template>
+    <template #footer>
+      <BtnMain
+        class="footer"
+        extend
+        @click="resolve"
+      >
+        {{ $t("ok") }}
```
Comment (peronczyk) [inline src/popup/components/Modals/ConsensusInfo.vue:42]: I suggest to use `text` property for regular string values.

File: src/popup/components/Modals/ConsensusInfo.vue [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      </BtnMain>
+    </template>
+  </Default>
+</template>
+
+<script>
+import {} from 'vue-i18n';
```
Comment (peronczyk) [inline src/popup/components/Modals/ConsensusInfo.vue:49]: I guess this is some kind of leftover.

File: src/popup/components/Modals/ConsensusInfo.vue [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+import Default from './Default.vue';
+import BtnMain from '../buttons/BtnMain.vue';
+import { BLOG_CLAIM_TIP_URL } from '../../utils/constants';
```
Comment (peronczyk) [inline src/popup/components/Modals/ConsensusInfo.vue:52]: Please put the utils/constants imports befor component imports.

File: src/popup/components/Modals/ConsensusInfo.vue [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    BtnMain,
+  },
+  props: {
+    resolve: { type: Function, required: true },
+    close: { type: Function, default: null },
+  },
+  data: () => ({
+    BLOG_CLAIM_TIP_URL,
+    UNFINISHED_FEATURES: process.env.UNFINISHED_FEATURES,
+  }),
```
Comment (peronczyk) [inline src/popup/components/Modals/ConsensusInfo.vue:68]: I suggest to use `setup()` hook.

File: src/popup/components/Modals/ConsensusInfo.vue [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+        extend
+        @click="resolve"
+      >
+        {{ $t("ok") }}
+      </BtnMain>
+    </template>
+  </Default>
+</template>
+
+<script>
```
Comment (peronczyk) [inline src/popup/components/Modals/ConsensusInfo.vue:48]: I strongly suggests to use TypeScript instead if JS.

File: src/popup/components/Modals/ConsensusInfo.vue [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+  },
+  data: () => ({
+    BLOG_CLAIM_TIP_URL,
+    UNFINISHED_FEATURES: process.env.UNFINISHED_FEATURES,
+  }),
+};
+</script>
+
+<style lang="scss" scoped>
+@use "../../../styles/share-info";
```
Comment (peronczyk) [inline src/popup/components/Modals/ConsensusInfo.vue:73]: This file is written in a really bad way - mostly because it applies styling onto html elements instead of using CSS classes. I strongly discourage you from importing it.

File: src/popup/components/Modals/ConsensusInfo.vue [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+@use "../../../styles/variables";
```
Comment (peronczyk) [inline src/popup/components/Modals/ConsensusInfo.vue:74]: No variables are used here.

File: src/popup/components/Modals/ConsensusInfo.vue [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+@use "../../../styles/typography";
+@use "../../../styles/mixins";
```
Comment (peronczyk) [inline src/popup/components/Modals/ConsensusInfo.vue:76]: No mixins are used here.

File: src/popup/components/Modals/ConsensusInfo.vue [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+
+.content {
```
Comment (peronczyk) [inline src/popup/components/Modals/ConsensusInfo.vue:78]: Please wrap all CSS rules with a class applied to the root element of the component.

File: src/popup/components/buttons/BtnHelp.vue [hunk 11]
```diff
@@ -23,12 +23,18 @@ export default defineComponent({
...
+  setup(props, { emit }) {
     const { openModal } = useModals();
 
-    async function showHelpModal() {
-      return openModal(MODAL_HELP, {
+    function showHelpModal() {
+      if (props.isCustomEmit) {
+        emit('help');
+        return;
+      }
```
Comment (peronczyk) [inline src/popup/components/buttons/BtnHelp.vue:35]: I see the problem you want to solve here. But maybe approaching it in a little bit different way would be better. What I suggest is:
1. Remove the `isCustomEmit` prop.
2. Fire `emit('help')` always (without wrapping it with `if` statement`
3. Open the `MODAL_HELP` only when there's a `msg` and/or `title` props.

What do you think?

File: src/popup/components/MultisigProposalConsensus.vue [hunk 12]
```diff
@@ -2,9 +2,17 @@
...
-      class="label"
+      class="consensus-label"
     >
-      <span> {{ $t('multisig.consensus') }} </span>
+      <div class="label">
+        <span> {{ $t("multisig.consensus") }} </span>
+        <BtnHelp
+          class="btn-help"
+          is-custom-emit
+          @help="openModal(MODAL_CONSENSUS_INFO)"
```
Comment (peronczyk) [inline src/popup/components/MultisigProposalConsensus.vue:12]: I suggest to create a custom method for this action to have a better TS support/validation.

File: src/popup/components/MultisigVaultCreateReview.vue [hunk 13]
```diff
@@ -39,6 +39,12 @@
       </DetailsItem>
 
       <DetailsItem :label="$t('multisig.consensus')">
+        <template #label>
+          <BtnHelp
+            is-custom-emit
+            @help="openModal(MODAL_CONSENSUS_INFO)"
```
Comment (peronczyk) [inline src/popup/components/MultisigVaultCreateReview.vue:45]: I suggest to create a custom method for this action to have a better TS support/validation.

File: src/popup/components/MultisigProposalConsensus.vue [hunk 14]
```diff
@@ -80,13 +96,11 @@ export default defineComponent({
...
-      activeMultisigAccount,
-    } = useMultisigAccounts({ store: root.$store });
-    const {
-      accounts,
-      isLocalAccountAddress,
-    } = useAccounts({
+    const { openModal } = useModals();
+    const { activeMultisigAccount } = useMultisigAccounts({
+      store: root.$store,
+    });
```
Comment (peronczyk) [inline src/popup/components/MultisigProposalConsensus.vue:102]: I don't see a reason for making this change.

File: src/popup/components/MultisigProposalConsensus.vue [hunk 15]
```diff
@@ -174,31 +188,41 @@ export default defineComponent({
...
 });
 </script>
 
 <style lang="scss" scoped>
-@use '../../styles/variables';
-@use '../../styles/typography';
-@use '../../styles/mixins';
+@use "../../styles/variables";
+@use "../../styles/typography";
+@use "../../styles/mixins";
```
Comment (peronczyk) [inline src/popup/components/MultisigProposalConsensus.vue:201]: I don't see a reason for making this change - we are using single apostrophes everywhere.

**Review by peronczyk — CHANGES_REQUESTED** [review]

File: src/popup/components/MultisigProposalConsensus.vue [hunk 16]
```diff
@@ -2,9 +2,17 @@
   <div class="transaction-multisig-consensus">
     <div
       v-if="!isPendingMultisigTxCompleted"
-      class="label"
+      class="consensus-label"
     >
-      <span> {{ $t('multisig.consensus') }} </span>
+      <div class="label">
+        <span> {{ $t("multisig.consensus") }} </span>
```
Comment (peronczyk) [inline src/popup/components/MultisigProposalConsensus.vue:8]: I suggest to remove the spaces around the translated string.

File: src/popup/pages/MultisigDetails.vue [hunk 17]
```diff
@@ -122,13 +131,15 @@ export default defineComponent({
...
+      openModal,
+      MODAL_CONSENSUS_INFO,
     };
   },
 });
 </script>
 
 <style scoped lang="scss">
-@use '../../styles/variables';
+@use "../../styles/variables";
```
Comment (peronczyk) [inline src/popup/pages/MultisigDetails.vue:142]: I don't see a reason for this change as we are using single apostrophes everywhere.

**Review by peronczyk — CHANGES_REQUESTED** [review]

File: src/popup/components/MultisigProposalConsensus.vue (hunk 16 above)
Comment (pstachel) [inline src/popup/components/MultisigProposalConsensus.vue:8]: Why did you change single quotes to double quotes?

File: src/popup/components/MultisigVaultCreateReview.vue [hunk 18]
```diff
@@ -103,9 +118,11 @@ import TokenAmount from './TokenAmount.vue';
 import ConsensusLabel from './ConsensusLabel.vue';
 
 import LoadingIcon from '../../icons/animated-spinner.svg?skip-optimize';
+import BtnHelp from './buttons/BtnHelp.vue';
```
Comment (pstachel) [inline src/popup/components/MultisigVaultCreateReview.vue:121]: Please order the imports correctly

File: src/popup/locales/en.json [hunk 19]
```diff
@@ -163,6 +163,14 @@
       "takingLong": "This step may take a few moments.",
       "syncing": "Syncing..."
     },
+    "consensus-info": {
```
Comment (pstachel) [inline src/popup/locales/en.json:166]: Please use camelCase in translations

File: src/popup/components/Modals/ConsensusInfo.vue [hunk 20]
```diff
@@ -0,0 +1,90 @@
...
+  },
+  props: {
+    resolve: { type: Function, required: true },
+    close: { type: Function, default: null },
+  },
+};
+</script>
+
+<style lang="scss" scoped>
+@use "../../../styles/typography";
```
Comment (pstachel) [inline src/popup/components/Modals/ConsensusInfo.vue:67]:
```suggestion
@use '../../../styles/typography';
```

File: src/popup/components/MultisigProposalConsensus.vue [hunk 21]
```diff
@@ -2,9 +2,17 @@
   <div class="transaction-multisig-consensus">
     <div
       v-if="!isPendingMultisigTxCompleted"
-      class="label"
+      class="consensus-label"
     >
-      <span> {{ $t('multisig.consensus') }} </span>
+      <div class="label">
+        <span> {{ $t('multisig.consensus') }} </span>
```
Comment (pstachel) [inline src/popup/components/MultisigProposalConsensus.vue:8]: You can remove the spacing around braces

File: src/popup/components/MultisigProposalConsensus.vue [hunk 22]
```diff
@@ -2,9 +2,17 @@
...
+        <BtnHelp
+          class="btn-help"
+          is-custom-emit
```
Comment (peronczyk) [inline src/popup/components/MultisigProposalConsensus.vue:11]: Can be removed

File: src/popup/components/MultisigVaultCreateReview.vue [hunk 23]
```diff
@@ -39,6 +39,12 @@
       </DetailsItem>
 
       <DetailsItem :label="$t('multisig.consensus')">
+        <template #label>
+          <BtnHelp
+            is-custom-emit
```
Comment (peronczyk) [inline src/popup/components/MultisigVaultCreateReview.vue:44]: Can be removed

**Review by pstachel — APPROVED** [review]

**Review by peronczyk — APPROVED** [review]

File: src/popup/components/buttons/BtnHelp.vue [hunk 24]
```diff
@@ -24,17 +24,21 @@ export default defineComponent({
...
-      return openModal(MODAL_HELP, {
-        icon: props.icon || 'info',
-        title: props.title,
-        msg: props.msg,
-        option: props.option,
-        textCenter: true,
-      });
+    function showHelpModal() {
+      emit('help');
+      console.log(props);
```
Comment (CedrikNikita) [inline src/popup/components/buttons/BtnHelp.vue:32]:
```suggestion
```

**Review by CedrikNikita — DISMISSED** [review]
